plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorker.java

```diff
@@ -412,7 +411,7 @@ private void scaleKubernetesClusterSize() throws CloudRuntimeException {
         } else { // upscale, same node count handled above
             scaleUpKubernetesClusterSize(newVmRequiredCount);
         }
-        kubernetesCluster = updateKubernetesClusterEntry(clusterSize, null);
+        kubernetesCluster = updateKubernetesClusterEntry(clusterSize, serviceOffering);
```

Comment:
Will this have any effect in some cases?
What happens if scale is called for both size and offering, the new size is lesser and for some reason offering scaling fails? Will it already update the cluster's CPU and memory based on the new offering?

Reply:
reverted this change @shwstppr This is not required as we are now correctly updating the existing cluster. Previously the core issue was because of updating the cluster wrongly.